bugwarrior/services/todoist.py

```diff
@@ -0,0 +1,279 @@
+                else:
+                    # if no timezone set is set remove tzinfo
+                    # otherwixe it will be treated as UTC by default
+                    todoist_due = self.record["due"]["date"].replace(tzinfo=None)
+            else:
+                # the due is just a `date` with no time or timezone.
+                todoist_due = datetime.combine(
+                    self.record["due"]["date"], default_time, tzinfo=None
+                )
+        else:
+            todoist_due = None
+
+        # deadline if set is only a date with no time or timezone.
+        todoist_deadline = (
+            datetime.combine(self.record["deadline"]["date"], default_time, tzinfo=None)
+            if self.record["deadline"]
+            else None
+        )
+
+        task = {
+            "project": self.extra["project"],
+            "priority": self.get_priority(),
+            "annotations": self.extra.get("annotations", []),
+            "tags": (
+                self.get_tags_from_labels(self.record["labels"])
+                if self.record["labels"]
+                else []
+            ),
+            "scheduled": None,
+            "due": todoist_due,
+            "status": "completed" if self.record["is_completed"] else "pending",
+            "entry": self.record["created_at"],
+            self.ID: self.record["id"],
+            self.CONTENT: self._unescape_content(self.record["content"]),
+            self.DESCRIPTION: self._unescape_content(self.record["description"]),
+            self.DUE: todoist_due,
+            self.DEADLINE: todoist_deadline,
+            self.DURATION: self.extra["duration"],
+            self.ASSIGNEE: self.extra["assignee"],
+            self.ASSIGNER: self.extra["assigner"],
+            self.SECTION: self.extra["section"],
+            self.URL: self.record["url"],
+        }
+        return task
+
+    def get_default_description(self):
+        description = self.build_default_description(
+            title=self._unescape_content(self.record["content"]),
+            url=self.record["url"],
+            number=self.record["id"],
+            cls="issue",
+        )
+        return description
+
+
+class TodoistService(Service):
+    ISSUE_CLASS = TodoistIssue
+    CONFIG_SCHEMA = TodoistConfig
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # apply additional filters
+        if self.config.only_if_assigned:
+            if self.config.also_unassigned:
+                filter = (
+                    self.config.filter
+                    + f" & (!assigned | shared & assigned to: {self.config.only_if_assigned})"
+                )
+            else:
+                filter = (
+                    self.config.filter
+                    + f" & (!shared | shared & assigned to: {self.config.only_if_assigned})"
+                )
+        else:
+            filter = self.config.filter
```

Comment:
I would find this easier to follow if it were a bit more concise. For example:
```suggestion
        filter = self.config.filter
        if self.config.only_if_assigned:
            assigned = "assigned" if self.config.also_unassigned else "shared"
            filter += f" & (!{assigned} | shared & assigned to: {self.config.only_if_assigned})"
```

Reply:
How about this.  The final filter is a bit more explicit to include both !shared and !assigned, but a bit clearer to follow.
```python
        # apply additional filters
        filter = self.config.filter
        if self.config.only_if_assigned:
            # fetch personal tasks (!shared)
            personal = "!shared"
            # fetch assigned tasks in shared projects (shared & assigned)
            shared_assigned = f"| shared & assigned to: {self.config.only_if_assigned}"
            # fetch unassigned tasks in shared projects (!assigned)
            unassigned = "| !assigned" if self.config.also_unassigned else ""
            filter += f" & ({personal} {shared_assigned} {unassigned})"
```

Reply:
Much better than the original or my suggestion, thanks! One question:

```py
            # fetch unassigned tasks in shared projects (!assigned)
            unassigned = "| !assigned" if self.config.also_unassigned else ""
```
I take it the `assigned` filter implicitly applies the `shared` filter, but would there be any harm in making it explicit to make the code more clear? (`"| shared & !assigned"`)